src/ShellCheck/Analytics.hs

```diff
@@ -3858,13 +3858,13 @@ groupByLink :: (a -> a -> Bool) -> [a] -> [[a]]
 groupByLink f list =
     case list of
         [] -> []
-        (x:xs) -> g x [] xs
+        (x:xs) -> foldr c n xs x []
   where
-    g current span (next:rest) =
+    c next rest current span =
```

Comment:
Why was this changed from `g` to `c`?

Reply:
Before, `g` handled both the cons and nil cases, since it was doing explicit recursion. Now that `foldr` is handling the recursion, there's two different functions, so I called them `c` for the cons case and `n` for the nil case.